clients/src/test/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImplTest.java

```diff
@@ -660,19 +688,16 @@ public void testSameAssignmentReconciledAgainWithMissingTopic() {
         );
         membershipManager.onHeartbeatRequestSent();
         assertEquals(MemberState.RECONCILING, membershipManager.state());
-        clearInvocations(membershipManager);
 
         // Receive extended assignment - assignment received but no reconciliation triggered
         membershipManager.onHeartbeatSuccess(createConsumerGroupHeartbeatResponse(assignment2).data());
         assertEquals(MemberState.RECONCILING, membershipManager.state());
-        verifyReconciliationNotTriggered(membershipManager);
```

Comment:
similar to comment above, I would keep this kind of verification (spying on the membershipMgr only). The membership mgr holds the state machine for the member as part of a group, it's all intertwined, we've seen how small changes in one membership func affects/breaks states and transitions, so all coverage we can maintain here is important.